tests/std/tests/P2165R4_tuple_like_tuple_members/test.cpp

```diff
@@ -0,0 +1,280 @@
+        static_assert(constructible_from<Tuple<const int&&>, const TupleLike<int>>);
+
+        auto get_letter      = [](size_t Idx) { return static_cast<char>('A' + Idx); };
+        TupleLike<TmpChar> a = {get_letter(Indices)...}; // 'A', 'B', 'C', ...
+
+        Tuple<TmpChar> t1(a);
+        assert((get<Indices>(t1) == get_letter(Indices) && get<Indices>(a) == get_letter(Indices)) && ...);
+
+        Tuple<TmpChar&> t2(a);
+        assert((&get<Indices>(t2) == &get<Indices>(a)) && ...);
+
+        Tuple<const TmpChar&> t3(a);
+        assert((&get<Indices>(t3) == &get<Indices>(a)) && ...);
+
+        Tuple<TmpChar> t4(as_const(a));
+        assert((get<Indices>(t4) == get_letter(Indices) && get<Indices>(a) == get_letter(Indices)) && ...);
+
+        Tuple<const TmpChar&> t5(as_const(a));
+        assert((&get<Indices>(t5) == &get<Indices>(a)) && ...);
+
+        Tuple<TmpChar> t6(std::move(a));
+        assert((get<Indices>(t6) == get_letter(Indices) && get<Indices>(a) == '\x7F') && ...);
+        ((get<Indices>(a) = std::move(get<Indices>(t6))), ...);
+
+        Tuple<const TmpChar&> p7(std::move(a));
+        assert((&get<Indices>(p7) == &get<Indices>(a)) && ...);
+
+        Tuple<TmpChar&&> t8(std::move(a));
+        assert((&get<Indices>(t8) == &get<Indices>(a)) && ...);
+
+        Tuple<const TmpChar&&> t9(std::move(a));
+        assert((&get<Indices>(t9) == &get<Indices>(a)) && ...);
+
+        Tuple<TmpChar> t10(std::move(as_const(a)));
+        assert((get<Indices>(t10) == get_letter(Indices) && get<Indices>(a) == get_letter(Indices)) && ...);
+
+        Tuple<const TmpChar&> t11(std::move(as_const(a)));
+        assert((get<Indices>(t11) == get_letter(Indices) && get<Indices>(a) == get_letter(Indices)) && ...);
+
+        Tuple<const TmpChar&&> t12(std::move(as_const(a)));
+        assert((get<Indices>(t12) == get_letter(Indices) && get<Indices>(a) == get_letter(Indices)) && ...);
+
+        // Check explicit specifier
+        static_assert(!convertible_to<string_view, string>);
+        static_assert(convertible_to<TupleLike<int>, Tuple<long>>);
+        static_assert(!convertible_to<TupleLike<string_view>, Tuple<string>> || N == 0);
+        static_assert(constructible_from<Tuple<string>, TupleLike<string_view>>);
+
+        return true;
+    }
+
+    template <size_t... Indices>
+    static constexpr bool test_tuple_like_allocator_constructor(index_sequence<Indices...>) {
+        // Test tuple(tuple-like, allocator_arg_t, Alloc)
+        static_assert(constructible_from<Tuple<string>, allocator_arg_t, allocator<char>, TupleLike<string>&>);
+        static_assert(constructible_from<Tuple<string>, allocator_arg_t, allocator<char>, const TupleLike<string>&>);
+        static_assert(constructible_from<Tuple<string>, allocator_arg_t, allocator<char>, TupleLike<string>>);
+        static_assert(constructible_from<Tuple<string>, allocator_arg_t, allocator<char>, const TupleLike<string>>);

```

Comment:
The comment mentions `tuple-like` as the first constructor argument, but the following `static_assert`s correctly place it as the last constructor argument.